crates/aptos-rosetta/src/types/objects.rs

```diff
@@ -625,144 +630,195 @@ fn parse_transfer_from_txn_payload(
+        );
     }
-    operations
+
+    Ok(operations)
 }
 
 fn parse_stakepool_changes(
+    version: u64,
     address: AccountAddress,
     data: &[u8],
     events: &[ContractEvent],
     operation_index: u64,
-) -> Vec<Operation> {
+) -> ApiResult<Vec<Operation>> {
     let mut operations = Vec::new();
-    // TODO: handle unwrap
-    let stakepool: StakePool = bcs::from_bytes(data).unwrap();
-
-    stakepool.set_operator_events.key();
-    if let Some(operator) = get_set_operator_from_event(events, stakepool.set_operator_events.key())
-    {
-        operations.push(Operation::set_operator(
-            operation_index,
-            Some(OperationStatusType::Success),
-            address,
-            operator,
-        ));
+    if let Ok(stakepool) = bcs::from_bytes::<StakePool>(data) {
+        stakepool.set_operator_events.key();
+        if let Some(operator) =
+            get_set_operator_from_event(events, stakepool.set_operator_events.key())
+        {
+            operations.push(Operation::set_operator(
+                operation_index,
+                Some(OperationStatusType::Success),
+                address,
+                operator,
+            ));
+        }
+    } else {
+        warn!(
+            "Failed to parse stakepool for {} at version {}",
+            address, version
+        );
     }
 
-    operations
+    Ok(operations)
 }
 
-fn parse_coinstore_changes(
+async fn parse_coinstore_changes(
+    coin_cache: Arc<CoinCache>,
+    coin_type: TypeTag,
+    version: u64,
     address: AccountAddress,
     data: &[u8],
     events: &[ContractEvent],
     mut operation_index: u64,
-) -> Vec<Operation> {
+) -> ApiResult<Vec<Operation>> {
     let coin_store: CoinStoreResource = bcs::from_bytes(data).unwrap();
     let mut operations = vec![];
 
-    // TODO: Handle other coins
-    if let Some(amount) = get_amount_from_event(events, coin_store.withdraw_events().key()) {
-        operations.push(Operation::withdraw(
-            operation_index,
-            Some(OperationStatusType::Success),
-            address,
-            native_coin(),
-            amount,
-        ));
-        operation_index += 1;
-    }
+    // Retrieve the coin type
+    let currency = coin_cache
+        .get_currency(coin_type.clone(), Some(version))
+        .await
+        .map_err(|err| {
+            ApiError::CoinTypeFailedToBeFetched(Some(format!(
+                "Failed to retrieve coin type {} {}",
+                coin_type, err
+            )))
+        })?;
+
+    // Skip if there is no currency that can be found
+    if let Some(currency) = currency {
+        if let Some(amount) = get_amount_from_event(events, coin_store.withdraw_events().key()) {
+            operations.push(Operation::withdraw(
+                operation_index,
+                Some(OperationStatusType::Success),
+                address,
+                currency.clone(),
+                amount,
+            ));
+            operation_index += 1;
```

Comment:
It's the order of all the operations, required by the spec.

It's only incremented for withdraws because there are two operations in this function.  First one, then the second one.